app/scripts/controllers/attachPVC.js

```diff
@@ -252,26 +250,14 @@ angular.module('openshiftConsole')
             $scope.alerts = {};
 
             // save deployment or deployment config
-            if (deploymentConfig) {
-              DataService.update("deploymentconfigs", deploymentConfig.metadata.name, deploymentConfig, context).then(
-                function() {
-                  $window.history.back();
-                },
-                function(result) {
-                  displayError("An error occurred attaching the persistent volume claim to deployment config.", getErrorDetails(result));
-                }
-              );
-            }
-            if (deployment) {
-              DataService.update("replicationcontrollers", deployment.metadata.name, deployment, context).then(
-                function() {
-                  $window.history.back();
-                },
-                function(result) {
-                  displayError("An error occurred attaching the persistent volume claim to deployment.", getErrorDetails(result));
-                }
-              );
-            }
+            DataService.update(resourceGroupVersion, resource.metadata.name, $scope.attach.resource, context).then(
+              function() {
+                $window.history.back();
+              },
+              function(result) {
+                displayError("An error occurred attaching the persistent volume claim to deployment config.", getErrorDetails(result));
```

Comment:
message says deployment config in it, should either be generic and say resource, or embed the lower-cased humanized kind here